transports/quic/tests/smoke.rs

```diff
@@ -27,13 +36,33 @@ async fn smoke_tls() -> Result<()> {
     smoke::<libp2p_quic::TlsCrypto>().await
 }
 
-async fn create_swarm<C: Crypto>(keylog: bool) -> Result<Swarm<RequestResponse<PingCodec>>>
+trait GenerateKeypair: Crypto {
+    fn generate_keypair() -> Self::Keypair;
+}
+
+#[cfg(feature = "noise")]
+impl GenerateKeypair for libp2p_quic::NoiseCrypto {
+    fn generate_keypair() -> Self::Keypair {
+        generate_noise_keypair()
+    }
+}
+
+#[cfg(feature = "tls")]
+impl GenerateKeypair for libp2p_quic::TlsCrypto {
+    fn generate_keypair() -> Self::Keypair {
+        generate_tls_keypair()
+    }
+}
```

Comment:
would it make sense to move this to the crypto trait? seems useful, and especially for ed25519_dalek, it's really annoying that you have to match the rand version. this makes sure that applications don't have to depend on a specific rand.

Reply:
I don’t think so. There is no need